packages/js/components/src/rich-text-editor/rich-text-editor.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { BlockEditorProvider } from '@wordpress/block-editor';
+import { BlockInstance } from '@wordpress/blocks';
+import { SlotFillProvider } from '@wordpress/components';
+import { debounce } from 'lodash';
+import {
+	createElement,
+	useCallback,
+	useEffect,
+	useState,
+	useRef,
+} from '@wordpress/element';
+import React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { ShortcutProvider } from '@wordpress/keyboard-shortcuts';
+
+/**
+ * Internal dependencies
+ */
+import { EditorWritingFlow } from './editor-writing-flow';
+import { registerBlocks } from './utils/register-blocks';
+
+registerBlocks();
+
+type RichTextEditorProps = {
+	blocks: BlockInstance[];
+	onChange: ( changes: BlockInstance[] ) => void;
+	entryId?: string;
+};
+
+export const RichTextEditor: React.VFC< RichTextEditorProps > = ( {
+	blocks,
+	onChange,
+} ) => {
+	const blocksRef = useRef( blocks );
+
+	const [ , setRefresh ] = useState( 0 );
+
+	useEffect( () => {
+		blocksRef.current = blocks;
+		// If there is a props change we need to update the ref and force re-render.
+		// Note: Because this component is memoized and because we don't re-render
+		// when this component initiates a change, a prop change won't force the re-render
+		// you'd expect. A change to the blocks must come from outside the editor.
+		setRefresh( ( r ) => r + 1 );
+	}, [ blocks ] );
+
+	// Use a combo of memoization and debounce to refresh every 200 milliseconds,
+	// ensuring that history is kept up to date for undo.
+	const debouncedRefresh = useCallback(
+		debounce( () => setRefresh( ( refresh ) => ( refresh += 1 ) ), 200 ),
```

Comment:
Why you are mutating the refresh var? Why `setRefresh( ( refresh ) => refresh + 1 )` does not satisfy the porpose of this?

Reply:
Good call.  I think it should be fine to update this.  This line is copied from the day one implementation, so hopefully it's not triggering some weird side effect, but I don't see any issues changing this right now.